flytekit/experimental/eager_function.py

```diff
@@ -127,7 +128,7 @@ async def __call__(self, **kwargs):
                 "If you need to use a subworkflow, use a static @workflow or nested @eager workflow."
             )
 
-        if self.ctx.execution_state.is_local_execution():
+        if not self.local_entrypoint and self.ctx.execution_state.is_local_execution():
```

Comment:
Maybe we just check if the remote is `None`?

||local execution | remote execution | 
|----| ----| ---|
|remote| Use flytekit remote to run the task | Use flytekit remote to run the task |
|no remote| Run the task function locally | Run the task in the driver (eager) pod |

Reply:
I think we'd want to run the AsyncEntity locally even if `remote is None` right? e.g.

```
# wf.py
@eager(remote=remote)
def eager_wf(): ...
```

```
pyflyte run wf.py eager_wf
```
should run locally, even if the user has configured a remote object.

It's only when they say `local_entrypoint=True` that the behavior changes such that it runs remotely on the configured remote.

Reply:
```python
@eager(remote=remote, local_entrypoint=False)
def wf()
   ...
```
qq: In the local execution, what happens in this case?, will we also use flytekit remote to execute the task?

Reply:
### Case 1:

Just specifying a `remote` shouldn't change any of the local behavior: it'll just run locally as async functions and ignore the `remote object.

```
@eager(remote=remote, local_entrypoint=False)
# pyflyte run: runs the eager workflow locally, runs the tasks/subworkflows locally
# pyflyte run --remote: runs the eager workflow remotely, runs the tasks/subworkflows remotely
```

### Case 2:

`local_entrypoint=True` just forces the eager workflow's tasks/subworkflows to run remotely (while running the eager workflow itself locally) as a debugging/development tool.
```
@eager(remote=remote, local_entrypoint=True)  # runs eager workflow
# pyflyte run: runs the eager workflow locally, but forces tasks and subworkflows to run remotely
# pyflyte run --remote: runs the 
```

Reply:
also happy to change the name `local_entrypoint` to something clearer. The use case is: "I want to run the eager workflow locally, but run the tasks/subworkflows remotely to that I can debug it, e.g. get access to intermediary results"

Reply:
Thanks for clarifying. I mean, we can change the default behaviour, like
```python
@eager(remote=remote)
# pyflyte run: runs the eager workflow locally, but forces tasks and subworkflows to run remotely
# pyflyte run --remote: runs the eager workflow remotely, runs the tasks/subworkflows remotely

@eager()
# If remote isn't set, 
# pyflyte run: runs the eager workflow locally, runs the tasks/subworkflows locally
```

Reply:
yes, if we want to add a new arg, it will be better to change `local_entrypoint` to something clearer.

Reply:
any suggestions?

Reply:
> ```
> @eager(remote=remote)
> # pyflyte run: runs the eager workflow locally, but forces tasks and subworkflows to run remotely. If fail to read > remote config, Fallback to local execution,
> # pyflyte run --remote: runs the eager workflow remotely, runs the tasks/subworkflows remotely
> ```

The problem with this is that I have to keep changing my code if I want to run completely locally vs. completely remotely. Think of `remote=remote` as just configuration, similar to the configuration that you give to `@task`. I'd like to run my eager workflow locally and remotely by just changing the `pyflyte run` call (adding `--remote` flag for remote execution).

The `local_entrypoint=True` flag is exclusively for the hybrid case of running the eager workflow locally but the tasks/subworkflows remotely.

If it makes the use case clearer we can just introduce a wrapper decorator to isolate this use case:

```
# this will always just run locally and you can't run this on a remote Flyte cluster
@local_entrypoint(remote=remote)
async def wf():
   ...
```

(btw I just took the naming from [modal](https://modal.com/docs/guide/ex/hello_world#running-it), open to suggestions for other names)

Reply:
Got it. Rather than a decorator, I prefer `@eager(remote=remote, local_entrypoint=True)`.
I'm not good at naming. Maybe we merge it first and then rename it once someone finds a better name.

Reply:
Cool, will make sure to document this well on flytesnacks